lib/devices/ios/ios-controller.js

```diff
@@ -1232,8 +1233,34 @@ iOSController.isLocked = function (cb) {
   cb(new NotYetImplementedError(), null);
 };
 
+var osaBackground = function (appName, secs, cb) {
+  logger.debug("Execute background osa: '" + appName + "'," + secs);
+  exec ('osascript ' + backgroundOSA + ' \'' + appName + '\' \'' + secs + '\'', function (error, stdout, stderr) {
+    if (error !== null) {
+      cb (error);
+    } else {
+      cb (null, {status:0, result: "" + stderr + ";" + stdout});
+    }
+  })
+}
+
 iOSController.background = function (secs, cb) {
-  this.proxy(["au.background(", secs, ")"].join(''), cb);
+  var v = (this.args.platformVersion || this.iOSSDKVersion);
+  if (v >= 9) {
+    if (!this.args.appName) {
+      this.getAppNameFromApp (function(err, appName) {
+        if (err) {
```

Comment:
This would be better as `if (err) return cb(err);`. That would be in line with our style. The call to `osaBackground` then does not need to be in an `else` block.